Source/Fuse.Nodes/Visual.Children.uno

```diff
@@ -29,24 +29,24 @@ namespace Fuse
 
 		public T FirstChild<T>() where T: Node
 		{
-			for (var i = 0; i < Children.Count; ++i)
+			var c = Children_first;
+			while (c != null)
 			{
-				var c = Children[i] as T;
-				if (c != null)
-					return c;
+				var v = c as T;
+				if (v != null) return v;
+				c = c.Children_next;
 			}
 			return null;
 		}
 
-		public void RemoveAllChildren<T>()
+		public void RemoveAllChildren<T>() where T: Node
 		{
-			var i = Children.Count - 1;
-			while (i >= 0)
+			var c = Children_first;
+			while (c != null)
 			{
-				if (Children[i] is T)
-					Children.RemoveAt(i);
-
-				i = Math.Min(i, Children.Count) - 1;
+				var v = c as T;
+				if (v != null) Children_Remove(v);
```

Comment:
Doesn't this imply the loop needs `Children_SafeForEach` or equivalent?  It's removing items from the list it is iterating.